my-app/src/components/Input.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+export default class Input extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { value: '' };
+    }
+
+    handleChange = (event) => {
+        this.setState({ value: event.target.value }, () => {
+            this.props.handleStatusChange({ [this.props.name]: !!this.state.value });
+        });
+    };
```

Comment:
to samo co wcześniej. trzymaj value i onChange w rodzicu i przekazuj jako props.
a ten komponent niech jest dumb